### PR TITLE
#12933 Windows open editor for shortcut file `.lnk`

### PR DESCRIPTION
**Pull request by ShadelessFox**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.navigator/OSGI-INF/l10n/bundle.properties [hunk 1]
```diff
@@ -65,6 +65,8 @@ command.org.jkiss.dbeaver.core.resource.link.folder.name=Link Folder
 command.org.jkiss.dbeaver.core.resource.link.folder.description=Create link to some folder on file system
 command.org.jkiss.dbeaver.core.sql.script.associate.name=Associate with data source
 command.org.jkiss.dbeaver.core.sql.script.associate.description=Associate selected script(s) with data source
+command.org.jkiss.dbeaver.core.show.in.explorer.name=Show resource in explorer
+command.org.jkiss.dbeaver.core.show.in.explorer.description=Opens system explore and highlights selected resource
```
Comment (flambchap) [inline plugins/org.jkiss.dbeaver.ui.navigator/OSGI-INF/l10n/bundle.properties]: explorer*

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java [hunk 2]
```diff
@@ -1427,6 +1428,27 @@ public static boolean launchProgram(String path)
...
+            final Map<String, String> params = Collections.singletonMap(ShowInSystemExplorerHandler.RESOURCE_PATH_PARAMETER, path);
+            final ParameterizedCommand pc = ParameterizedCommand.generateCommand(command, params);
+
+            try {
+                handlerService.executeCommand(pc, null);
+            } catch (Exception e) {
+                DBWorkbench.getPlatformUI().showError("Error showing path in system explorer", "Can't show '" + path + "' in system explorer", e);
+            }
+        }
+    }
```
Comment (flambchap) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java]: I believe these methods belong to RuntimeUtils. Moreover, it's not exactly clear why we need two distinct methods.

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreResources.properties [hunk 3]
```diff
@@ -508,3 +508,6 @@ dialog_connection_edit_wizard_error_md5_title = Error making MD5
 dialog_connection_edit_wizard_error_md5_msg =Cannot generate password hash
 ## Connection edit ##
 
+## Resource Handlers
+resource_shortcut_deleted_title = Error opening shortcut
+resource_shortcut_deleted_message = The item ''{0}'' that this shortcut refers to has been deleted.\n\nDo you want to delete this shortcut?
```
Comment (flambchap) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreResources.properties:513]: which*

**Review by flambchap — CHANGES_REQUESTED** [review]

**Review by flambchap — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by serge-rider** [comment]
Let's keep `showInSystemExplorer` function in the handler. UIUtils is already too messy.

**Comment by ShadelessFox** [comment]
I've added it to `UIUtils` because it can be used in other places too. For example: ERD image export, data transfer into file, etc.

**Pull request merged**